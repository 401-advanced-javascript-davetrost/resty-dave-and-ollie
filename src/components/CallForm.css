.CallForm {
  input[name="url"] {
    display: block;
    width: 80%;
  }

  input[name="method"] {
    display: none;
  }

  span:hover {
    background-color: #CCCCCC;
  }

  span:active {
    background-clip: #2D5986;
  }

  input[type=radio]:checked + span {
    background-color: #2D5986;
    color: white;
  }

  span {
    font-family: Verdana;
    font-size: 16px;
    margin: 5px;
    padding: 5px;
  }
}