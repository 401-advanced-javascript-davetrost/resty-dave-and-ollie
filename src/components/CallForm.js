import React, { Component } from 'react';
// import PropTypes from 'prop-types';
import styles from './CallForm.css';

export default class CallForm extends Component {
  state = {

  }

  // handleSubmit = event => {
  //   event.preventDefault();

  //   this.setState(state => {
  //     console.log(state);
  //   });
  // }

  render() {
    return (
      <form className={styles.CallForm}>
        <input type="text" name="url" placeholder="URL"></input>
        <label>
          <input defaultChecked name="method" type="radio" value="get" />
          <span>GET</span>
        </label>
        <label>
          <input name="method" type="radio" value="post" />
          <span>POST</span>
        </label>
        <label>
          <input name="method" type="radio" value="put" />
          <span>PUT</span>
        </label>
        <label>
          <input name="method" type="radio" value="patch" />
          <span>PATCH</span>
        </label>
        <label>
          <input name="method" type="radio" value="delete" />
          <span>DELETE</span>
        </label>
        <label>
          <button type="submit">Go</button>
        </label>

        <textarea name="requestBody" maxLength="250" minLength="200" placeholder="Raw JSON Body"></textarea>
      </form>
    );
  }

  static propTypes = {

  }
}


