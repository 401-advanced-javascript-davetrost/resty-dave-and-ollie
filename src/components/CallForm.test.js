import React from 'react';
import { shallow } from 'enzyme';
import CallForm from './CallForm';

describe('CallForm Component', () => {
  it('renders callForm', () => {
    const wrapper = shallow(<CallForm
      url="abc123"
      method="get"
      body="heyarnold.com/api/characters"
      handleUrlChange={() => true}
      handleMethodChange={() => true}
      handleBodyChange={() => true}
      handleSubmit={() => true}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
