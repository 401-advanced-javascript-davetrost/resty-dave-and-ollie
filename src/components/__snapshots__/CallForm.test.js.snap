// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallForm Component renders callForm 1`] = `
<form
  className="CallForm"
>
  <input
    name="url"
    onChange={[Function]}
    placeholder="URL"
    type="text"
    value="abc123"
  />
  <label>
    <input
      checked={true}
      name="method"
      onChange={[Function]}
      type="radio"
      value="get"
    />
    <span>
      GET
    </span>
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="post"
    />
    <span>
      POST
    </span>
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="put"
    />
    <span>
      PUT
    </span>
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="patch"
    />
    <span>
      PATCH
    </span>
  </label>
  <label>
    <input
      checked={false}
      name="method"
      onChange={[Function]}
      type="radio"
      value="delete"
    />
    <span>
      DELETE
    </span>
  </label>
  <label>
    <button
      onClick={[Function]}
      type="submit"
    >
      Go
    </button>
  </label>
  <textarea
    maxLength="250"
    minLength="200"
    name="requestBody"
    onChange={[Function]}
    placeholder="Raw JSON Body"
    value="heyarnold.com/api/characters"
  />
</form>
`;
